src/main/java/org/spongepowered/common/mixin/core/server/MixinPlayerList.java

```diff
@@ -540,6 +543,10 @@ public EntityPlayerMP recreatePlayerEntity(EntityPlayerMP entityPlayerMP, int ta
         this.playerEntityList.add(entityPlayerMP);
         entityPlayerMP.addSelfToInternalCraftingInventory();
 
+        // Update reducedDebugInfo game rule
```

Comment:
Probably not. It's not necessary in Vanilla because all worlds use the same game rules. On Sponge however, the worlds can have different settings for `reducedDebugInfo`.